public/app/features/explore/hooks/useStateSync/migrators/v1.ts

```diff
@@ -16,16 +16,33 @@ export interface ExploreURLV1 extends BaseExploreURL {
 
 export const v1Migrator: MigrationHandler<ExploreURLV0, ExploreURLV1> = {
   parse: (params) => {
+    let panesFmt = params.panes?.toString();
     if (!params || !params.panes || typeof params.panes !== 'string') {
-      return {
+      const defaultObj: ExploreURLV1 = {
         schemaVersion: 1,
         panes: {
           [generateExploreId()]: DEFAULT_STATE,
         },
       };
+      // some URL builders will want to put multiple panes in multiple params. If that's the case here, combine them into one object
+      if (params && params.panes && Array.isArray(params.panes)) {
```

Comment:
@Elfo404 Just wondering what you think about this. The URL builder utility we have builds our panes a little differently than we currently accept them. Since it does return a valid URL, I feel like it's more of something we should support in the parser but I can do something else if youd like.

Basically, the [builder](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/utils/url.ts#L66) will take multiple values and repeat the parameter instead of combining them. This means we end up with something like `explore?panes={'abc':{}}&panes={'def':{}}` when we want `explore?panes={'abc':{}, 'def':{}}`. 

Should we use a different function or is this change to this parser okay?

Reply:
where is this builder used?

the explore URL structure is now well defined, i don't think that just because a function generates URL in a wrong way we should support them, we should probably fix that function(or how it is used) instead.

if we expect the `panes` param to be a keyed object we should probably not pass it as an array in https://github.com/grafana/grafana/pull/77336/files#diff-287b3efbe1ba71d24789e1853d050e5208211141d6ac2f46f5488f1095abdf1dR62, i don't think the builder is doing anything wrong, we are simply generating the "panes" param wrongly.

Reply:
@Elfo404 The builder is used here https://github.com/grafana/grafana/pull/77336/files#diff-287b3efbe1ba71d24789e1853d050e5208211141d6ac2f46f5488f1095abdf1dR62

Reply:
according to [our own documentation](https://grafana.com/docs/grafana/latest/explore/#generating-explore-urls-from-external-tools) about explore URL format:

>panes is a url-encoded JSON object of panes, where each key is the pane ID and each value is an object matching the following schema:

i'm not saying that using query parameter in that way is wrong in absolute, but that our own API uses a single object with keyed entries, and instead of adapting the API to accept this specific format because of how we implemented the functionality in this PR, we can simply use the expected format in the first place.

>to store multiple items in an array is relatively common
also to nitpick, we are not storing multiple items of an array, we are storing multiple entries of an object that then we recombine at parse time. it is not wrong, but it just complicates the parsing logic for no real benefit